projects/tuist/fixtures/app_with_docc/Project.swift

```diff
@@ -0,0 +1,30 @@
+import ProjectDescription
+
+let project = Project(
+    name: "App",
+    organizationName: "tuist.io",
+    targets: [
+        Target(
+            name: "App",
+            platform: .iOS,
+            product: .app,
+            bundleId: "io.tuist.framework",
+            deploymentTarget: .iOS(targetVersion: "13.0", devices: .iphone),
+            infoPlist: .default,
+            sources: ["Targets/App/Sources/**"],
+            dependencies: [.project(target: "Framework", path: ".")]
+        ),
+        Target(
+            name: "Framework",
+            platform: .iOS,
+            product: .framework,
+            bundleId: "io.tuist.framework",
+            deploymentTarget: .iOS(targetVersion: "13.0", devices: .iphone),
+            infoPlist: .default,
+            sources: [SourceFileGlob(
+                "Targets/Framework/Sources/**",
+                excluding: "Targets/Framework/Sources/ImportantDocumentation.docc/**/*.swift"
```

Comment:
I think tuist should be smart enough to exclude thos files automatically, as they are part of a .docc archive

Reply:
This one could be related to the above comment, but when removing the excluding parameter (while the `.docc` archive is still considered a valid Source File) a warning is emitted because I believe it doesn't get the same special treatment as a Folder (`xcasset`) would. 

Is there a way to replicate that type of excluding behavior for certain types of Source files? 🤔 

Example output:
```
Generating workspace App.xcworkspace
Generating project App
Trying to add a file at path /Users/3051989/Documents/workspace/personal/tuist/projects/tuist/fixtures/app_with_docc/Targets/Framework/Sources/ImportantDocumentation.docc/Tutorial/Tutorials/Tutorial1/Code.swift to a build phase that hasn't been added to the project.
Consider creating an issue using the following link: https://github.com/tuist/tuist/issues/new/choose
Program ended with exit code: 1
```

Reply:
I wonder if we can at the point of unfolding the globs filter out the nested files - today we have a few special cases for `xcassets` 🤐 

e.g.

```swift
// BuildPhaseGenerator.swift
let isWithinAssets = pathString.contains(".xcassets/") || pathString.contains(".scnassets/")

/// Assets that are part of a .xcassets or .scnassets folder
/// are not added individually. The whole folder is added
/// instead as a group.
if isWithinAssets {
     return
}
```

This doesn't scale very well to more use cases 😞

Reply:
Thank you for sharing that code snippet @kwridan! 

I was able to add this capability by filtering out the Swift tutorial files in the `generateSourcesBuildPhase` 👍